File: scripts/extract-pptx.py
import zipfile
import re
import xml.etree.ElementTree as ET
from pathlib import Path

pptx = Path(r"D:\CurrentAug032025\Platform\BillionTechWebsite\docs\BillionTech_BigBasket_Proposal_v2.pptx")
out = Path(r"D:\CurrentAug032025\Platform\BillionTechWebsite\docs\_extracted_bigbasket.txt")

NS = "{http://schemas.openxmlformats.org/drawingml/2006/main}t"

with zipfile.ZipFile(pptx) as z:
    slides = sorted(
        [n for n in z.namelist() if n.startswith("ppt/slides/slide") and n.endswith(".xml")],
        key=lambda x: int(re.search(r"slide(\d+)", x).group(1)),
    )
    chunks = []
    for s in slides:
        root = ET.fromstring(z.read(s))
        texts = [t.text for t in root.iter(NS) if t.text and t.text.strip()]
        chunks.append(f"=== {s} ===\n" + "\n".join(texts))
    out.write_text("\n\n".join(chunks), encoding="utf-8")
    print(f"Wrote {out} ({len(slides)} slides)")

File: scripts/extract-pptx.mjs
import fs from 'node:fs';
import path from 'node:path';
import { fileURLToPath } from 'node:url';
import JSZip from 'jszip';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const pptxPath = path.join(__dirname, '..', 'docs', 'BillionTech_BigBasket_Proposal_v2.pptx');
const outPath = path.join(__dirname, '..', 'docs', '_extracted_bigbasket.txt');

const buf = fs.readFileSync(pptxPath);
const zip = await JSZip.loadAsync(buf);
const slideNames = Object.keys(zip.files)
  .filter((n) => /^ppt\/slides\/slide\d+\.xml$/.test(n))
  .sort((a, b) => Number(a.match(/slide(\d+)/)[1]) - Number(b.match(/slide(\d+)/)[1]));

const chunks = [];
for (const name of slideNames) {
  const xml = await zip.files[name].async('string');
  const texts = [...xml.matchAll(/<a:t[^>]*>([^<]*)<\/a:t>/g)].map((m) => m[1]).filter(Boolean);
  chunks.push(`=== ${name} ===\n${texts.join('\n')}`);
}
fs.writeFileSync(outPath, chunks.join('\n\n'), 'utf8');
console.log(`Wrote ${outPath} (${slideNames.length} slides)`);
